rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -208,16 +208,46 @@ Node::register_param_change_callback(CallbackT && callback)
 }
 
 template<typename ParameterT>
+void
+Node::set_parameter_if_not_set(
+  const std::string & name,
+  const ParameterT & default_value)
+{
+  rclcpp::parameter::ParameterVariant parameter_variant;
+  if (!this->get_parameter(name, parameter_variant)) {
+    this->set_parameters({
+          rclcpp::parameter::ParameterVariant(name, default_value),
+        });
```

Comment:
looks weird/over-indented but I guess it's what it takes to make uncrustify happy...